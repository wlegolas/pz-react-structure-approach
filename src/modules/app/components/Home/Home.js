import React from 'react'

const Home = ({ children }) => {
  return <div className="pz-container">{children}</div>
}

export default Home
