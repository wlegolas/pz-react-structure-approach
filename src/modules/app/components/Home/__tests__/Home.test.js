import toJson from 'enzyme-to-json'
import React from 'react'
import Home from '../Home'

describe('<Home />', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const wrapper = mount(
        <Home>
          <div className="my-content">My Content</div>
        </Home>
      )

      expect(toJson(wrapper)).to.matchSnapshot()
    })

    it('should render a nested component', () => {
      const NestedComponent = () => (
        <div className="nested-cmp">
          <h1>Nested component</h1>
        </div>
      )
      const wrapper = mount(
        <Home>
          <NestedComponent />
        </Home>
      )

      expect(wrapper.find('.nested-cmp')).to.have.lengthOf(1)
    })
  })
})
