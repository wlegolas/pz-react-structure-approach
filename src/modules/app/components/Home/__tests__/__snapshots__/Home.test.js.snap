// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render() should render the component 1`] = `
<Home>
  <div
    className="pz-container"
  >
    <div
      className="my-content"
    >
      My Content
    </div>
  </div>
</Home>
`;
